gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/src/DbDataStoreUtils.js

```diff
@@ -486,109 +480,244 @@ gpii.oauth2.dbDataStore.findAuthByCodePostProcess = function (data) {
  */
-gpii.oauth2.dbDataStore.doRevokeClientCredentialsToken = function (saveDataSource, clientCredentialsTokenRecord) {
+gpii.oauth2.dbDataStore.addAuthorization = function (saveDataSource, authorizationType, authorizationData) {
     var promiseTogo = fluid.promise();
+    var data, error;
+
+    // Verify the authorization type
+    if (authorizationType !== gpii.oauth2.docTypes.gpiiAppInstallationAuthorization &&
+        authorizationType !== gpii.oauth2.docTypes.onboardedSolutionAuthorization &&
+        authorizationType !== gpii.oauth2.docTypes.privilegedPrefsCreatorAuthorization &&
+        authorizationType !== gpii.oauth2.docTypes.webPrefsConsumerAuthorization) {
+        error = gpii.oauth2.composeError(gpii.oauth2.errors.unauthorized, {docName: authorizationType});
+        promiseTogo.reject(error);
+        return promiseTogo;
+    }
 
-    if (clientCredentialsTokenRecord === undefined) {
-        promiseTogo.resolve(undefined);
-    } else {
-        var data = fluid.copy(clientCredentialsTokenRecord);
-        data.revoked = true;
-        promiseTogo = gpii.oauth2.dbDataStore.updateRecord(saveDataSource, gpii.oauth2.dbDataStore.docTypes.clientCredentialsToken, "id", data);
+    // The authorization data must exist
+    if (!authorizationData) {
+        error = gpii.oauth2.composeError(gpii.oauth2.errors.missingDoc, {docName: authorizationType});
+        promiseTogo.reject(error);
+        return promiseTogo;
     }
 
-    return promiseTogo;
-};
-// ==== End of revokeClientCredentialsToken()
+    // Construct the authorization data structure based on the authorization type
+    if (authorizationType === gpii.oauth2.docTypes.gpiiAppInstallationAuthorization) {
+        data = {
+            clientId: authorizationData.clientId,
+            gpiiToken: authorizationData.gpiiToken,
+            accessToken: authorizationData.accessToken,
+            revoked: false,
+            timestampCreated: gpii.oauth2.getCurrentTimestamp(),
+            timestampRevoked: null,
+            timestampExpires: authorizationData.timestampExpires
+        };
+    }
 
-/**
- * A post process function for implementing findAuthByClientCredentialsAccessToken() API.
- * @param data {Object} An object in a structure of:
- * {
- *     doc: {
- *         oauth2ClientId: {String}
- *     },
- *     value: {
- *         allowAddPrefs: {Boolean}
- *     }
- * }
- * @return {Object} The object is in the structure of:
- * {
- *     oauth2ClientId: {String},
- *     allowAddPrefs: {Boolean}
- * }
- * If the given parameter is not in an expected structure, `undefined` is returned.
- */
-gpii.oauth2.dbDataStore.findAuthByClientCredentialsAccessTokenPostProcess = function (data) {
-    if (data.doc && data.value) {
-        return {
-            oauth2ClientId: data.doc.oauth2ClientId,
-            allowAddPrefs: data.value.allowAddPrefs
+    if (authorizationType === gpii.oauth2.docTypes.onboardedSolutionAuthorization) {
```

Comment:
I think that it would be better here to structure the `authorizationType` tests with `if ... else ...` rather than independent `if` statements. So that it is clear that they are mutually exclusive and that you don't need to read through the test conditions to see if multiple might apply.